### PR TITLE
Linker test

### PR DESCRIPTION
**Pull request by lewing**
Add some linker failure cases

### CODE REVIEW

File: mcs/tools/linker/Tests/System.Runtime.Serialization/test-dcs-01.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	public static void Main (string []args) {
+		var source = new List<string>();
+		source.Add("a");
+		source.Add("b");
+		using (System.IO.MemoryStream stream = new System.IO.MemoryStream()) {
+			var serializer = new System.Runtime.Serialization.DataContractSerializer(typeof(List<string>));
+			serializer.WriteObject(stream, source);
+			stream.Flush();
+
+			Console.WriteLine (stream.ToArray().ToString());
```
Comment (marek-safar) [inline mcs/tools/linker/Tests/System.Runtime.Serialization/test-dcs-01.cs]: Is this line needed?

File: mcs/tools/linker/Makefile [hunk 2]
```diff
@@ -22,7 +22,9 @@ TEST_CASES := \
 	mscorlib/test-string-03.cs \
 	mscorlib/test-task-01.cs \
 	System/test-security.cs \
-	System.Core/test-plinq-01.cs
+	System.Core/test-plinq-01.cs \
+	System.Core/test-queryable-01.cs \
+	System.Runtime.Serialization/test-dcs-01.cs
```
Comment (marek-safar) [inline mcs/tools/linker/Makefile]: You'll have to add rules with new assembly references

File: mcs/tools/linker/Tests/System.Runtime.Serialization/test-dcs-01.cs (hunk 1 above)
Comment (lewing) [reply mcs/tools/linker/Tests/System.Runtime.Serialization/test-dcs-01.cs]: removed

**Comment by lewing** [comment]
@monojenkins build failed

**Pull request merged**